### DISCUSSION TITLE
How about Adding/Overloading Some Operators for Signal to Make Signal Operations More Readable

### DISCUSSION THREAD

**Discussion by Lazy-Rabbit-2001** [godotengine/godot-proposals#8046] [Scripting]
(I don't know if hers is a good place to talk about it, nor do i make it clear that if someone has mentioned this idea before)
As is known to all, signals are now belong to a native class Signal, which allows us to directly connect them to methods (Callables).
However, most of times we are prefer to use a operator to connect signal, just like how it is done in C#.
My thought is to add some operators like  "==>" to be meant as "connect()[without flags]", "=/=" as "disconnect()" and "=?=" as "is_connected()".  
For example
```GDScript
signal my_signal
...
my_signal.connect(my_func)

# Replaced by operators
# Equals "my_signal.connect(my_func)"
my_signal += my_func 
#or
my_signal => my_func # Returns an int

# Disconnection
# Equals "my_signal.disconnect(my_func)"
my_signal =/=> my_func 
#or
my_signal -= my_func # Returns nothing

# Check
# Equals "my_signal.is_connected(my_func)"
print(my_signal =?=> my_func) 
#or
my_signal == my_func # Returns True/False, here it's obviously 'false'
```
If we need to pass in a flag, we can add a method `bind_flags()` for a Signal
```GDScript
# Equals "my_signal.connect(my_func, ...)"
my_signal.bind_flags(...) => my_func
```
When it comes to signal emission, we can use an operator `++` to mean "emit", and add a method `bind_params()` to make it emitted with parameters
```GDScript
# Equals "my_signal.emit(1, 2, 3)"
my_signal.bind_params(1, 2, 3) ++
```
Developers are still allowed to use methods to connect, disconnect, emit and check signals

How do you like this idea? Please share yours in this post :)

**Comment by dalexeev** [comment]
We prefer keywords and methods rather than symbolic operators. There are a few exceptions (`$Node`, `%UniqueNode`, `&"StringName"`, etc.), but they are due to frequency of use and shortening the code (no quotes and parentheses). In this case, this does not reduce the time to write the code (due to autocompletion), but it significantly worsens the readability.

Perhaps the `+=` and `-=` shortcuts for `connect()` and `disconnect()` look ok to those coming from C#, but the rest of the operators look confusing to me and won't be useful for other data types.

**Reply by Lazy-Rabbit-2001** [reply]
Maybe `+=` and `-=` could be friendly for those developing with C#. As for `is_connected()`, it's very difficult to think about an operator.
Anyway thanks for your comment :)

**Comment by AThousandShips** [comment]
Please only open either a discussion or a issue for a proposal, closing this as the issue has more information/discussion
